src/blocks.rs

```diff
@@ -0,0 +1,188 @@
+            "gray_terracotta",
+            "gray_terracotta",
+            "gray_terracotta",
+            "gray_terracotta",
+        ],
+        is_solid: true,
+    },
+    Block {
+        name: "light_gray_terracotta",
+        texture_names: [
+            "light_gray_terracotta",
+            "light_gray_terracotta",
+            "light_gray_terracotta",
+            "light_gray_terracotta",
+            "light_gray_terracotta",
+            "light_gray_terracotta",
+        ],
+        is_solid: true,
+    },
+    Block {
+        name: "orange_terracotta",
+        texture_names: [
+            "orange_terracotta",
+            "orange_terracotta",
+            "orange_terracotta",
+            "orange_terracotta",
+            "orange_terracotta",
+            "orange_terracotta",
+        ],
+        is_solid: true,
+    },
+    Block {
+        name: "red_terracotta",
+        texture_names: [
+            "red_terracotta",
+            "red_terracotta",
+            "red_terracotta",
+            "red_terracotta",
+            "red_terracotta",
+            "red_terracotta",
+        ],
+        is_solid: true,
+    },
+    Block {
+        name: "terracotta",
+        texture_names: [
+            "terracotta",
+            "terracotta",
+            "terracotta",
+            "terracotta",
+            "terracotta",
+            "terracotta",
+        ],
+        is_solid: true,
+    },
+    Block {
+        name: "yellow_terracotta",
+        texture_names: [
+            "yellow_terracotta",
+            "yellow_terracotta",
+            "yellow_terracotta",
+            "yellow_terracotta",
+            "yellow_terracotta",
+            "yellow_terracotta",
+        ],
+        is_solid: true,
+    },
+];
+
+pub static AIR: u8 = 0;
+pub static GRASS: u8 = 1;
+pub static DIRT: u8 = 2;
+pub static STONE: u8 = 3;
+pub static BEDROCK: u8 = 4;
+pub static RED_SAND: u8 = 5;
+pub static BROWN_TERRACOTTA: u8 = 6;
+pub static CYAN_TERRACOTTA: u8 = 7;
+pub static GRAY_TERRACOTTA: u8 = 8;
+pub static LIGHT_GRAY_TERRACOTTA: u8 = 9;
+pub static ORANGE_TERRACOTTA: u8 = 10;
+pub static RED_TERRACOTTA: u8 = 11;
+pub static TERRACOTTA: u8 = 12;
+pub static YELLOW_TERRACOTTA: u8 = 13;
+
+pub fn get_texture_uvs(texture_name: &str) -> Option<[f32; 2]> {
+    match texture_name {
+        "stone" => Some([0.0, 0.0]),
+        "dirt" => Some([0.25, 0.0]),
+        "grass_top" => Some([0.5, 0.0]),
+        "grass_side" => Some([0.75, 0.0]),
+        "bedrock" => Some([0.0, 0.25]),
+        "red_sand" => Some([0.25, 0.25]),
+        "brown_terracotta" => Some([0.5, 0.25]),
+        "cyan_terracotta" => Some([0.75, 0.25]),
+        "gray_terracotta" => Some([0.0, 0.5]),
+        "light_gray_terracotta" => Some([0.25, 0.5]),
+        "orange_terracotta" => Some([0.5, 0.5]),
+        "red_terracotta" => Some([0.75, 0.5]),
+        "terracotta" => Some([0.0, 0.75]),
+        "yellow_terracotta" => Some([0.25, 0.75]),
+        _ => None,
+    }
+}
```

Comment:
Why not add the texture coordinates directly to the `Block`? You can still have a mapping function if you make it a `const fn`.

Reply:
The texture names are shared between blocks. I think I could namespace the mapping function under Block. Something like this?
```rs
Block::texture_uvs(texture_name: &str) -> Option<[f32; 2]>
block.uvs -> [[f32; 2]; 6]
```

Reply:
I could also imagine something like this:
```Rust
type Texture = [f32; 2];

struct Block {
    pub textures: [Texture; 6],
    pub is_solid: bool,
}

const DirtTexture: Texture = [0.25, 0.0];

const BLOCKS: [Block; 1] = [
    Block {
        textures: [DirtTexture; 6],
        is_solid: true,
    }
];
```

But I noticed that you have texture names without actual textures ("air" for example).

Reply:
From a size efficiency: on a 64 bit system `&str` is a pointer and a length, so 16 bytes in total. `[f32; 2]` is just 8 bytes, so storing the textures directly instead of the name is even more space efficient.